### PR TITLE
Release upper bound on scikit-learn

### LINKED ISSUE #1350: CI Failing due to scikit-learn 1.4
#### Describe the bug

The latest release of `scikit-learn` (v1.4) is causing issues in the `MetricFrame` tests.

#### Steps/Code to Reproduce

See GitHub Actions

#### Expected Results

Green builds

#### Actual Results

Red builds

#### Screenshots

N/A

#### Versions

Current `main`

### PR DESCRIPTION
**Pull request by riedgar-ms**
## Description

We had temporarily added an upper bound on SciKit-Learn since there were issues with some of our tests. Remove the bound and address the test failures.

Closes #1350 

## Tests

- [ ] no new tests required
- [ ] new tests added
- [x] existing tests adjusted

## Documentation

- [x] no documentation changes needed
- [ ] user guide added or updated
- [ ] API docs added or updated
- [ ] example notebook added or updated

### CODE REVIEW

**Review by romanlutz — APPROVED** [review]

**Review by adrinjalali — APPROVED** [review]
lol, this is one way to do it. But we need to fix our metric issue and pass labels around.

**Comment by adrinjalali** [comment]
We need to fix the MetricFrame issue though, since as it is one cannot really use it with a bunch of sklearn metrics.

**Comment by riedgar-ms** [comment]
Can you clarify what you mean by 'pass labels around' @adrinjalali ? The warnings were being thrown by [balanced accuracy score](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.balanced_accuracy_score.html) and that doesn't have a `labels=` argument.

**Comment by adrinjalali** [comment]
Now I can't seem to find it, I thought I had an issue for it. 

The issue is that MetricFrame creates subgroups of data, and calculates the metrics for those subgroups. However, it's very likely that those subgroups don't have all the possible labels in their y_true or y_pred, and that for some sklearn metrics (like roc_auc_score (https://scikit-learn.org/dev/modules/generated/sklearn.metrics.roc_auc_score.html#sklearn.metrics.roc_auc_score) causes and issue.

So we need to pass labels around.

**Comment by riedgar-ms** [comment]
@adrinjalali  But that's what I meant - the specific warning being suppressed is for balanced accuracy score. And that doesn't have a `labels=` argument. We only require that a metric have `y_true` and `y_pred` arguments.
